research/src/components/checkbox-anatomy.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import './anatomy.css'
+
+export default () => {
+  return (
+    <div className="component-anatomy">
+      <host>
+        <part name="label">
+          <slot>{'Default'}</slot>
+        </part>
+        <part name="checkbox">
```

Comment:
Also, I noticed that there is no slot in here? How are you expecting the user of this checkbox to adjust the content/colors, etc?

Reply:
My understanding is the "part" convention denotes an element selectable via CSS, similar to how a pseudo is targetable. I'm not sure there is a need to provide *content* to this element, though I think there is an argument for providing content for the indicators themselves.

I'm not really fond of this name because of the redundancy. Functionally this represents the container for the checkbox control itself. This complication arises because, in this model (and in many component libraries), the "checkbox" is a checkbox + label pairing. In FAST we've adjusted this to be "control", but perhaps there is a better name? Happy to hear ideas here.

Reply:
I get your overall premise with regards to the parts, but it's making an assumption of what the checkbox will contain and how it will want to be styled/modified. If everything that anyone could do with a checkbox could be done solely in CSS then a part would be sufficient. If someone needs potentially additional layers for particle effects or some other additive design effect they shouldn't be inhibited from doing so. The slot doesn't need to be leveraged but should be possible.

Reply:
@gregwhitworth can you help me understand your recommendation? 

Are you proposing that an app author can slot arbitrary content *inside* this element or are you proposing that the entire element implementation can be whole-sale replaced. In my mind, the later would *also* remove the checked/indeterminate indicators because they are structurally children of the implementation.

Reply:
With slots for both indicators, I also wasn't quite sure what it would mean to had a slot for the entire checkox. I'm also not sure that you can keep the slots for the indicators if you do that. So, I'd be worried that we would increase the complexity of customization for the most common cases.

Just thinking out loud here...by combining Houdini custom paints via CSS with CSS parts, is there anything that this configuration really can't handle that absolutely needs the "everything" slot?